Expense & room rate features

kavindiyamalki03-crypto opened the pull request:
Fixed expense API endpoint and tested locally.

Changes:
- Updated expense_service.dart: /expense → /expenses (plural)
- Tested all expense features with local backend 
- Reverted config.dart to production URL
- Resolved merge conflicts with dev

Related: Backend PR Zonova-Tech/ZonovaMistAPI#28 (merged)

Note: Backend needs to be deployed to production for this to work.

kavindiyamalki03-crypto committed:
room rate feature

kavindiyamalki03-crypto committed:
room rate page connect with database

kavindiyamalki03-crypto committed:
room rate page table for add changes

kavindiyamalki03-crypto committed:
room rate table for add the dilog box

kavindiyamalki03-crypto committed:
Added my updates from personal repo

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

kavindiyamalki03-crypto committed:
room rate page add

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

kavindiyamalki03-crypto committed:
Add expense add feature

Added expense service and UI screens

kavindiyamalki03-crypto committed:
Added room rate validations (Task 2 fixes)

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

kavindiyamalki03-crypto committed:
add some update for room screen

kavindiyamalki03-crypto committed:
Reconnect project

kavindiyamalki03-crypto committed:
Save current work before merging dev branch

kavindiyamalki03-crypto committed:
Commit remaining changes

kavindiyamalki03-crypto committed:
Resolved merge conflicts between dev and room_rate_new

kavindiyamalki03-crypto committed:
update app drawer file

kavindiyamalki03-crypto committed:
add some changes

kavindiyamalki03-crypto committed:
Fixed: Use AppConfig for API base URLs

kavindiyamalki03-crypto committed:
Added ExpenseCategory enum for hotel expense categories

kavindiyamalki03-crypto committed:
Removed test and debug comments

kavindiyamalki03-crypto committed:
Removed custom image handling code as per code review

kavindiyamalki03-crypto committed:
Revert "Removed custom image handling code as per code review"

This reverts commit db922d6a2cfef76798c50ad062b30e387ad0e22b.

kavindiyamalki03-crypto committed:
Reverted UI changes in login_screen and logic changes in common_image_manager as per code review

kavindiyamalki03-crypto committed:
expense page update

kavindiyamalki03-crypto committed:
login page update

kavindiyamalki03-crypto committed:
Revert "login page update"

This reverts commit fca306c0bf99dd8de8960c6ded895d4c31c4c868.

kavindiyamalki03-crypto committed:
Revert "expense page update"

This reverts commit 5be91672ed0623ed2d0d0cfa08385c4faa3564f7.

kavindiyamalki03-crypto committed:
Update rooms screen and plugin registrant

kavindiyamalki03-crypto committed:
Fix expense API endpoint URL - change /expense to /expenses

kavindiyamalki03-crypto committed:
Resolve merge conflicts with dev

kavindiyamalki03-crypto committed:
Fix expense API endpoint and revert to production URL

SameeraRukshan committed:
Merge branch 'dev' into room_rate_new

SameeraRukshan commented on lib/src/core/api/dashboard_service.dart:
```diff
@@ -244,24 +244,24 @@ class DashboardService {
     }).toList();
   }
 
-  /// Helper method to safely parse numeric values
```
keep all meaningful comments like these

SameeraRukshan commented on android/app/build.gradle.kts:
```diff
@@ -4,7 +4,11 @@ import java.io.FileInputStream
 plugins {
     id("com.android.application")
     id("org.jetbrains.kotlin.android")
+<<<<<<< HEAD
```
remove all generated merge conflict codes

SameeraRukshan commented on lib/services/expense_service.dart:
```diff
@@ -0,0 +1,145 @@
+  static String get baseUrl => '${AppConfig.apiBaseUrl}/expenses';
+
+  static final cloudinary = CloudinaryPublic(
+    'dqi0bndrs',
```
if this is a password get it from config file

SameeraRukshan commented on lib/src/features/auth/login_screen.dart:
```diff
@@ -1,14 +1,16 @@
+import 'dart:convert';
```
remove all changes in login screen

SameeraRukshan commented on macos/Flutter/GeneratedPluginRegistrant.swift:
```diff
@@ -22,5 +23,6 @@ func RegisterGeneratedPlugins(registry: FlutterPluginRegistry) {
   FlutterSecureStoragePlugin.register(with: registry.registrar(forPlugin: "FlutterSecureStoragePlugin"))
   JustAudioPlugin.register(with: registry.registrar(forPlugin: "JustAudioPlugin"))
   PathProviderPlugin.register(with: registry.registrar(forPlugin: "PathProviderPlugin"))
+  SharedPreferencesPlugin.register(with: registry.registrar(forPlugin: "SharedPreferencesPlugin"))
```
if this plugin is not used remove this

SameeraRukshan commented on pubspec.yaml:
```diff
@@ -1,37 +1,24 @@
-# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
-# In Windows, build-name is used as the major, minor, and patch parts
-# of the product and file versions while build-number is used as the build suffix.
+
```
revert all changes

kavindiyamalki03-crypto committed:
Fix code formatting in staff screens

kavindiyamalki03-crypto committed:
Reverse API base URL logic as per review

kavindiyamalki03-crypto committed:
Add helpful comments in dashboard_service

kavindiyamalki03-crypto committed:
Remove unnecessary comment from login screen

kavindiyamalki03-crypto committed:
resolve comments

kavindiyamalki03-crypto committed:
remove unnecsery comments

kavindiyamalki03-crypto committed:
resolve comments and fix bracket issue

kavindiyamalki03-crypto committed:
Resolve merge conflict in build.gradle.kts

SameeraRukshan commented on lib/src/core/api/dashboard_service.dart:
```diff
@@ -80,7 +80,7 @@ class DashboardService {
   }
 
-  /// Fetch revenue comparison data
+  /// Fetch expense comparison data
```
should be revenue right?

SameeraRukshan commented on lib/src/core/api/dashboard_service.dart:
```diff
@@ -251,7 +252,7 @@ class DashboardService {
     if (value is String) return double.tryParse(value) ?? 0.0;
     if (value is Map) {
-      // Handle MongoDB Decimal128 format: {"$numberDecimal": "123.45"}
+
```
revert

kavindiyamalki03-crypto committed:
Keep meaningful comment as requested in review

kavindiyamalki03-crypto committed:
Revert versionCode to 10 as requested in review

kavindiyamalki03-crypto committed:
Revert config.dart to original version as requested in review

kavindiyamalki03-crypto committed:
Remove all changes in login screen as requested in review

kavindiyamalki03-crypto committed:
Comment Photos line in common_image_manager as requested

kavindiyamalki03-crypto committed:
Revert dashboard_provider to dev branch version as requested

kavindiyamalki03-crypto committed:
Move Cloudinary cloud name to config file as requested

kavindiyamalki03-crypto committed:
resolve code content

kavindiyamalki03-crypto committed:
reverse code

kavindiyamalki03-crypto committed:
Revert login_screen to dev branch version as requested

kavindiyamalki03-crypto committed:
rivert comment

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

ShanPramuditha16 commented on lib/src/core/api/dashboard_service.dart:
Don't delete code comments

kavindiyamalki03-crypto replied:
fixed

ShanPramuditha16 commented on lib/src/features/home/dashboard/dashboard_screen.dart:
dont delete code comments

ShanPramuditha16 commented on lib/src/features/home/rooms/edit_room_screen.dart:
Verify the import paths for these 
import '../../core/api/api_service.dart';
import '../../core/auth/rooms_provider.dart';

ShanPramuditha16 reviewed (changes requested):
I’m requesting a few changes before this can be merged:

- Please do not remove any existing comments.
- Add clear and meaningful comments to the new/updated code where necessary.
- Verify that all import paths are correct and consistent.
- Make sure the application is fully tested locally before submitting the PR.

Once these updates are done, feel free to re-request a review. Thanks!

kavindiyamalki03-crypto committed:
Update base URL to new API endpoint (comments preserved)

kavindiyamalki03-crypto committed:
Update base URL in config.dart

kavindiyamalki03-crypto committed:
Merge remote changes and resolve conflicts

kavindiyamalki03-crypto committed:
resolve code comment and change base URL

kavindiyamalki03-crypto committed:
Restore deleted comments

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

ShanPramuditha16 reviewed (changes requested):
This PR has the same issues mentioned in the previous review.

Requested changes were not addressed:
- Ensure the application is tested locally before opening a PR.
- Attach screenshots of the application after testing to confirm functionality.
- Existing comments were removed — please do not remove any comments.
- Add proper comments to the code where required.
- Verify that all import paths are correctly set.

Please make these changes and update the PR before requesting another review.

kavindiyamalki03-crypto committed:
Restore deleted comments in main.dart

kavindiyamalki03-crypto committed:
Merge branch 'room_rate_new' of https://github.com/Zonova-Tech/ZonovaMist into room_rate_new

kavindiyamalki03-crypto committed:
Restore deleted comments in main.dart

kavindiyamalki03-crypto committed:
Revert add_staff_screen to dev branch version

kavindiyamalki03-crypto committed:
Revert staff screens to dev branch versions

kavindiyamalki03-crypto committed:
Revert todo_provider to dev branch version

kavindiyamalki03-crypto committed:
Revert main.dart to match dev exactly

kavindiyamalki03-crypto committed:
Revert dashboard_service to dev branch version

kavindiyamalki03-crypto committed:
Add proper comments to room_rate_page

kavindiyamalki03-crypto committed:
Add proper comments to room_rate_page and add_expense_page

kavindiyamalki03-crypto committed:
Add proper comments to room rate and expense pages

kavindiyamalki03-crypto committed:
Add proper comments

kavindiyamalki03-crypto committed:
Add proper comments

kavindiyamalki03-crypto committed:
Add proper comments to room rate and expense features

kavindiyamalki03-crypto committed:
Revert common_image_manager to dev branch version

kavindiyamalki03-crypto committed:
Update dashboard_service base URL to new API endpoint

kavindiyamalki03-crypto committed:
Revert app_router changes to original state

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

ShanPramuditha16 reviewed (changes requested):
<img width="1295" height="115" alt="Screenshot 2026-01-05 110718" src="https://github.com/user-attachments/assets/3ceaecb8-d989-4d7e-8178-602c08258a37" />
<img width="1291" height="987" alt="Screenshot 2026-01-05 110949" src="https://github.com/user-attachments/assets/2469ef9e-0488-4248-bc59-cd4698f43813" />

Requesting changes.

Please handle the following issues:
- Fix the existing errors that are causing build/runtime failures.
- The expense functionality is not working — unable to add expenses.

Ensure the issues are resolved, test the app locally, and update the PR before requesting another review.

kavindiyamalki03-crypto committed:
fix: downgrade font_awesome_flutter for Dart SDK compatibility

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

ShanPramuditha16 reviewed (changes requested):
Build Failure

This PR introduces a compilation error due to duplicate declarations of _renderBaseUrl in lib/config.dart.
The same constant is defined twice in the same scope, which causes the app to fail at build time:
_renderBaseUrl is already declared in this scope

Required Action
Remove the duplicate declaration or
Refactor configuration handling (e.g., environment-based URLs or separate config files)
Please address this issue and update the PR.

yusri009 committed:
used dio instead of http

kavindiyamalki03-crypto committed:
Merge company fixes for expense feature

kavindiyamalki03-crypto committed:
fix: remove duplicate _renderBaseUrl declaration

kavindiyamalki03-crypto commented:
@ShanPramuditha16 
Fixed the duplicate _renderBaseUrl declaration issue. Build should pass now.

kavindiyamalki03-crypto committed:
Merge branch 'dev' into room_rate_new

ShanPramuditha16 reviewed (changes requested):
Editing expenses is not working, while other functionalities work as expected.
Request: Please fix the expense edit functionality and ensure changes are properly saved and reflected.

kavindiyamalki03-crypto committed:
Fix: Expense edit functionality - preserve amount and update UI

- edit_expense_page.dart: Preserve original amount when editing
- view_expense_page.dart: Add API call and proper data return
- expenses_list_page.dart: Handle updates from ViewPage edit
- Fixed amount preservation in all edit scenarios
- Fixed UI auto-update without manual refresh

kavindiyamalki03-crypto committed:
Merge branch 'room_rate_new' of https://github.com/Zonova-Tech/ZonovaMist into room_rate_new

kavindiyamalki03-crypto commented:
@ShanPramuditha16 
Fixed! The expense edit functionality now works correctly in all scenarios.

 What was fixed:
- Amount preservation : Amount field now correctly preserved during edits
- Edit from ViewPage : Pencil icon edit now works properly with API integration
- UI auto-updates : List automatically updates without manual refresh needed

 Technical changes:
- `edit_expense_page.dart`: Extract and preserve original amount value
- `view_expense_page.dart`: Convert to ConsumerWidget, add proper API integration
- `expenses_list_page.dart`: Handle async updates from ViewPage edits

 Testing completed:
 Swipe edit button - Working, amount preserved  
 ViewPage pencil icon - Working, amount preserved  
 UI auto-refresh - Working without manual refresh  
 New expense creation - Amount correctly saved  

Ready for review and merge!

ShanPramuditha16 reviewed (approved):
looking good

ShanPramuditha16 committed:
Update meta and test_api dependencies in pubspec.lock

Bump meta from 1.16.0 to 1.17.0 and test_api from 0.7.6 to 0.7.7 in pubspec.lock to use the latest versions.
